1.0.0rc2: test_saveframe fails

When testing version 1.0.0 rc2 during build time, I see the following failure on the `test_saveframe` and `test_saveframe_arr` tests (Heliocentric/HeliographicCarrington/HeliographicStonyhurst/Helioprojective/…):

```
_______________________________ test_saveframe[Heliocentric] ________________________________

coordframe_scalar = <Heliocentric Coordinate (obstime=2018-01-01T00:00:00.000, observer=<HeliographicStonyhurst Coordinate for 'earth'>): (x, y, z) in arcsec
    [(0.1466223, 0.14395401, 0.2031206)]>
tmpdir = local('/tmp/pytest-of-pbuilder/pytest-0/test_saveframe_Heliocentric_0')

    @skip_windows_asdf
    def test_saveframe(coordframe_scalar, tmpdir):
        tree = {'frame': coordframe_scalar}
>       assert_roundtrip_tree(tree, tmpdir)

sunpy/io/special/asdf/tags/tests/test_coordinate_frames.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib/python3/dist-packages/asdf/tests/helpers.py:174: in assert_roundtrip_tree
    _assert_roundtrip_tree(*args, **kwargs)
/usr/lib/python3/dist-packages/asdf/tests/helpers.py:189: in _assert_roundtrip_tree
    with asdf.open(buff, mode='rw', extensions=extensions) as ff:
/usr/lib/python3/dist-packages/asdf/asdf.py:1350: in open_asdf
    ignore_missing_extensions=ignore_missing_extensions)
/usr/lib/python3/dist-packages/asdf/asdf.py:709: in _open_impl
    ignore_missing_extensions=ignore_missing_extensions)
/usr/lib/python3/dist-packages/asdf/asdf.py:662: in _open_asdf
    tree = yamlutil.tagged_tree_to_custom_tree(tree, self, _force_raw_types)
/usr/lib/python3/dist-packages/asdf/yamlutil.py:280: in tagged_tree_to_custom_tree
    return treeutil.walk_and_modify(tree, walker)
/usr/lib/python3/dist-packages/asdf/treeutil.py:206: in walk_and_modify
    return recurse(top)
/usr/lib/python3/dist-packages/asdf/treeutil.py:137: in recurse
    val = recurse(val)
/usr/lib/python3/dist-packages/asdf/treeutil.py:137: in recurse
    val = recurse(val)
/usr/lib/python3/dist-packages/asdf/treeutil.py:137: in recurse
    val = recurse(val)
/usr/lib/python3/dist-packages/asdf/treeutil.py:158: in recurse
    result = callback(result)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

node = {'data': <SphericalRepresentation (lon, lat, distance) in (deg, deg, AU)
    (0., -3.00275389, 0.98330099)>, 'frame_attributes': {'obstime': <Time object: scale='utc' format='isot' value=2018-01-01T00:00:00.000>}}

    def walker(node):
        if force_raw_types:
            return node
    
        tag_name = getattr(node, '_tag', None)
        if tag_name is None:
            return node
    
        tag_type = ctx.type_index.from_yaml_tag(ctx, tag_name)
        # This means the tag did not correspond to any type in our type index.
        if tag_type is None:
            if not ctx._ignore_unrecognized_tag:
                warnings.warn("{} is not recognized, converting to raw Python "
>                   "data structure".format(tag_name), AsdfConversionWarning)
E               asdf.exceptions.AsdfConversionWarning: tag:astropy.org:astropy/coordinates/frames/heliographic_stonyhurst-1.0.0 is not recognized, converting to raw Python data structure

/usr/lib/python3/dist-packages/asdf/yamlutil.py:251: AsdfConversionWarning
----------------------------------- Captured stderr call ------------------------------------
/usr/lib/python3/dist-packages/asdf/schema.py:227: UserWarning: Unable to locate schema file for 'tag:astropy.org:astropy/coordinates/frames/heliocentric-1.0.0': 'file:///usr/lib/python3/dist-packages/astropy/io/misc/asdf/schemas/astropy.org/astropy/coordinates/frames/heliocentric-1.0.0.yaml'
  warnings.warn(msg.format(tag, schema_path))
/usr/lib/python3/dist-packages/asdf/schema.py:227: UserWarning: Unable to locate schema file for 'tag:astropy.org:astropy/coordinates/frames/heliographic_stonyhurst-1.0.0': 'file:///usr/lib/python3/dist-packages/astropy/io/misc/asdf/schemas/astropy.org/astropy/coordinates/frames/heliographic_stonyhurst-1.0.0.yaml'
  warnings.warn(msg.format(tag, schema_path))
```

ASDF version: 2.3.2
Astropy version: 3.1.2
Python: 3.7.3

--- a/sunpy/io/special/asdf/tags/tests/test_coordinate_frames.py
+++ b/sunpy/io/special/asdf/tags/tests/test_coordinate_frames.py
@@ -7,6 +7,7 @@ import pytest
 import astropy.units as u
 from asdf.tests.helpers import assert_roundtrip_tree
 
+from sunpy.tests.helpers import asdf_entry_points
 from sunpy.coordinates.frames import (Heliocentric, HeliographicCarrington,
                                       HeliographicStonyhurst, Helioprojective)
 
@@ -16,6 +17,7 @@ asdf = pytest.importorskip('asdf', '2.0.2')
 @pytest.fixture(
     params=(Heliocentric, HeliographicCarrington, HeliographicStonyhurst,
             Helioprojective))
+@asdf_entry_points
 def coordframe_scalar(request):
     frame = request.param
 
@@ -30,6 +32,7 @@ def coordframe_scalar(request):
 @pytest.fixture(
     params=(Heliocentric, HeliographicCarrington, HeliographicStonyhurst,
             Helioprojective))
+@asdf_entry_points
 def coordframe_array(request):
     frame = request.param
 
@@ -50,12 +53,14 @@ skip_windows_asdf = pytest.mark.skipif(
 
 
 @skip_windows_asdf
+@asdf_entry_points
 def test_saveframe(coordframe_scalar, tmpdir):
     tree = {'frame': coordframe_scalar}
     assert_roundtrip_tree(tree, tmpdir)
 
 
 @skip_windows_asdf
+@asdf_entry_points
 def test_saveframe_arr(coordframe_array, tmpdir):
     tree = {'frame': coordframe_array}
     assert_roundtrip_tree(tree, tmpdir)

--- a/sunpy/io/special/asdf/tags/tests/test_genericmap.py
+++ b/sunpy/io/special/asdf/tags/tests/test_genericmap.py
@@ -13,6 +13,7 @@ from asdf.tests.helpers import assert_roundtrip_tree  # noqa
 
 import sunpy.map  # noqa
 from sunpy.data.test import get_test_filepath  # noqa
+from sunpy.tests.helpers import asdf_entry_points
 from sunpy.io.special.asdf.extension import SunpyExtension  # noqa
 
 
@@ -31,6 +32,7 @@ skip_windows_asdf = pytest.mark.skipif(
 
 
 @skip_windows_asdf
+@asdf_entry_points
 def test_genericmap_basic(aia171_test_map, tmpdir):
 
     tree = {'smap': aia171_test_map}
@@ -39,6 +41,7 @@ def test_genericmap_basic(aia171_test_map, tmpdir):
 
 
 @skip_windows_asdf
+@asdf_entry_points
 def test_genericmap_mask(aia171_test_map, tmpdir):
 
     mask = np.zeros_like(aia171_test_map.data)

--- a/sunpy/tests/helpers.py
+++ b/sunpy/tests/helpers.py
@@ -1,7 +1,8 @@
 import os
-import platform
 import urllib
+import platform
 import warnings
+import pkg_resources
 from functools import wraps
 
 import matplotlib.pyplot as plt
@@ -10,7 +11,8 @@ from matplotlib.testing import compare
 
 from sunpy.tests import hash
 
-__all__ = ['skip_windows', 'skip_glymur', 'skip_ana', 'warnings_as_errors']
+__all__ = ['skip_windows', 'skip_glymur', 'skip_ana',
+           'warnings_as_errors', 'asdf_entry_points']
 
 # SunPy's JPEG2000 capabilities rely on the glymur library.
 # First we check to make sure that glymur imports correctly before proceeding.
@@ -35,6 +37,11 @@ else:
 skip_windows = pytest.mark.skipif(platform.system() == 'Windows', reason="Windows")
 skip_glymur = pytest.mark.skipif(SKIP_GLYMUR, reason="Glymur can not be imported")
 skip_ana = pytest.mark.skipif(SKIP_ANA, reason="ANA is not available")
+
+
+# Skip if the SunPy ASDF entry points are missing.
+asdf_entry_points = pytest.mark.skipif(not list(pkg_resources.iter_entry_points('asdf_extensions', 'sunpy')),
+                                       reason="No SunPy ASDF entry points.")
 
 
 @pytest.fixture

